types/time.go

```diff
@@ -775,7 +776,15 @@ func ParseDateFormat(format string) []string {
 		}
 	}
 
-	seps = append(seps, format[start:])
+	var end = 0
+	for end = start; end < len(format) && !isValidSeparator(format[end], len(seps)); end++ {
+	}
+
+	end = mathutil.Min(end, len(format))
```

Comment:
IIRC golang builtin provides `math.Min` already, and `github.com/cznic/mathutil` is archived.

Reply:
@tisonkun You mean the built-in math.Min?, it only has double64 parameter method, correct me if i'am wrong